src/asciifile.rs

```diff
@@ -58,6 +134,81 @@ use std::ops::Deref;
 pub struct Position {
     pub row: usize,
     pub col: usize,
+    byte_offset: usize,
+}
+
+impl Position {
+    pub fn get_line<'m>(
+        &self,
+        file: &AsciiFile<'m>,
+        max_context_length_before: usize,
+        max_context_length_after: usize,
+    ) -> (LineContext, &'m str, LineContext) {
+        let (truncated_before, start) = AsciiFile::get_line_start_idx(
+            file.mapping,
+            self.byte_offset,
+            max_context_length_before,
+        );
+        let (truncated_after, end) =
+            AsciiFile::get_line_end_idx(file.mapping, self.byte_offset, max_context_length_after);
```

Comment:
A `Position` obtained from a different `AsciiFile` might cause the `debug_assert` in `get_line_end_idx to panic`.